### PR TITLE
new task: auto-tag products with incoming inventory

### PR DESCRIPTION
**Pull request by tekhaus**
Development notes:
- The paginated query loop on inventory level change is for handling the maximum of 100 variants in conjunction with 100 locations (i.e. inventory levels). Technically, the Shopify location limit is 20 for PLUS shops, unless they specifically request more. I have seen 100 in the wild.

### CODE REVIEW

**Comment by tekhaus** [comment]
[https://mechanic.canny.io/task-requests/p/auto-tag-incoming-products](https://trello.com/c/LbPzMxa3/45-https-mechaniccannyio-task-requests-p-auto-tag-incoming-products)

File: docs/auto-tag-products-with-incoming-inventory/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Auto-tag products with incoming inventory
+
+Tags: Auto-Tag, Bulk, Inventory, Products
```
Comment (mattsodomsky) [inline docs/auto-tag-products-with-incoming-inventory/README.md:3]: add Transfer tag, you'll need to add here as well: https://github.com/lightward/mechanic-tasks/blob/main/lib/task_schema.json

File: docs/auto-tag-products-with-incoming-inventory/script.liquid [hunk 2]
```diff
@@ -0,0 +1,260 @@
+{% comment %}
+  New task: Auto-tag products with incoming inventory
```
Comment (mattsodomsky) [inline docs/auto-tag-products-with-incoming-inventory/script.liquid:2]: Remove comment :)

**Review by mattsodomsky — CHANGES_REQUESTED** [review]
@tekhaus awesome work :) This is an exciting task!! A few very minor touchups and I'll get this merged.

**Review by mattsodomsky — APPROVED** [review]
Looks great! thank you!

**Pull request merged**